cmd/lakectl/cmd/abuse.go

```diff
@@ -23,21 +23,21 @@ var abuseCmd = &cobra.Command{
 	Hidden: true,
 }
 
-func readLines(path string) ([]string, error) {
+func readLines(path string) (lines []string, err error) {
 	reader := OpenByPath(path)
+	defer func() {
+		err = reader.Close()
```

Comment:
This is subtle and I believe when I discussed it with @itaiad200 he strongly asked that we  *not* do this sort of thing.  (You can put the body of the function after opening `reader` in a sub-function, get an error from there, and then close inline -- if you want to keep some elegance).

Also, while elegant, it can hide a *first* error (e.g. flagged by `scanner.Err()`) by returning only the *second* error (if `reader.Close()` _also_ fails).  I definitely prefer returning the first or ideally both:
```suggestion
		if closeErr = reader.Close(); closeErr != nil {
			if err == nil {
				err = closeErr
			} else {
				err = fmt.Errorf("%w, and while closing %s", err, closeErr)
			}				
```

Reply:
Agree, I'll align the above to use this practice.

Reply:
Updated the code - had to keep the named return values, without it the err value was back to nil after the return.
If it is too much I can just add DieErr as OpenByPath and the return function when using this helper does it.
what do you think?